### PR TITLE
feat: hide joined tables

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #4596

### Description:

Acceptance criteria
- [x] I'm able to hide joined tables from an explore ( doesn't show in the explore sidebar )
- [x] The fields of the hidden table should not be available in the filters
- [x] The fields of the hidden table can be used in fields SQL from other tables

Screenshot 2023-11-24 at 16 45 29

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/common/src/compiler/exploreCompiler.ts [hunk 1]
```diff
@@ -160,16 +164,20 @@ export class ExploreCompiler {
...
-                                }),
+                                (prevMetrics, metricKey) => {
+                                    const metric =
+                                        tables[join.table].metrics[metricKey];
+                                    return {
+                                        ...prevMetrics,
+                                        [metricKey]: {
+                                            ...metric,
+                                            hidden:
+                                                !!join.hidden || metric.hidden,
```
Comment (ZeRego) [inline packages/common/src/compiler/exploreCompiler.ts:175]: changed the function so I could access `metric.hidden` cleanly

File: docs/docs/references/joins.md [hunk 2]
```diff
@@ -228,3 +228,19 @@ models:
 ```
 
 Check out our [dimensions](/references/dimensions) and [metrics](/references/metrics) reference docs to see all of the other configurations you can use with your joined fields.
+
+## Hidde join tables
```
Comment (rephus) [inline docs/docs/references/joins.md:232]:
```suggestion
## Hide join tables
```

File: examples/full-jaffle-shop-demo/dbt/models/schema.yml [hunk 3]
```diff
@@ -1,6 +1,13 @@
 version: 2
 models:
   - name: customers
+    meta:
+      joins:
+        - join: membership
```
Comment (rephus) [inline examples/full-jaffle-shop-demo/dbt/models/schema.yml:6]: very cool

**Review by rephus — APPROVED** [review]
Looks good, nice data on jaffle shop

image

The `membership` still appears on the main explore, I'm curious, is there a way to hide it from there without breaking the join from customers ?( I know this is not in scope for this)

image

**Comment by ZeRego** [comment]
> The membership still appears on the main explore, I'm curious, is there a way to hide it from there without breaking the join from customers ?( I know this is not in scope for this)

It could be done by tagging the models in the yml and then filter in project table configuration.

**Pull request merged**